src/components/ListItem/ListItem.tsx

```diff
@@ -58,6 +58,10 @@ export interface ListItemProps extends VibeComponentProps {
    */
   tabIndex?: number;
   "data-testid"?: string;
+  /**
+   * The index of the item in the list
+   */
+  index?: number;
```

Comment:
just use the `id` prop and pass `${listId}-item-${index}` instead of introducing the `index` prop